### PR TITLE
Alphabetize the elements in the spec

### PR DESCRIPTION
**Pull request by jdmgoogle**
Alphabetize the elements in the spec (with two small changes).

Updated the style guide to reflect this ordering.

### CODE REVIEW

**Comment by pstenbjorn** [comment]
LGTM

File: STYLEGUIDE.md [hunk 1]
```diff
@@ -111,14 +111,25 @@ For example--
 ### Element/Attribute Ordering
 
 Within a type, elements should be listed first, followed by attributes. Each
-should be in alphabetical order:
+should be in alphabetical order, with only two exceptions:
+
+1. If an element named `Type` or `FooType` is an enumeration which contains a value
+   `other`, then `OtherType` or `OtherFooType` should be placed after `Type`.
```
Comment (cjerdonek) [inline STYLEGUIDE.md]: I would simplify this to say if an element named `Foo` ... then `OtherFoo` ....  The reason is that doesn't really matter if the type name ends in "Type" (and it needn't end that way).

**Comment by cjerdonek** [comment]
(Sorry for the duplicate comment, it disappeared on GitHub then came back.)  Aside from that one comment, LGTM.

**Comment by jdmgoogle** [comment]
@cjerdonek Updated.

Merging now.

**Pull request merged**